src/Facility.Definition/ServiceDefinitionError.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// An error while processing a service definition.
+	/// </summary>
+	public sealed class ServiceDefinitionError
+	{
+		/// <summary>
+		/// Creates an error.
+		/// </summary>
+		public ServiceDefinitionError(string error, NamedTextPosition position = null, Exception innerException = null)
+		{
+			if (error == null)
+				throw new ArgumentNullException(nameof(error));
+
+			Error = error;
+			Position = position;
+			InnerException = innerException;
+		}
+
+		/// <summary>
+		/// The error message.
+		/// </summary>
+		public string Error { get; }
+
+		/// <summary>
+		/// The position where the error took place, if any.
```

Comment:
Is the position ever unspecified?

Reply:
There is one place it isn't, but that can be changed.